File: rust/392.is-subsequence.rs
/*
 * @lc app=leetcode id=392 lang=rust
 *
 * [392] Is Subsequence
 */

// @lc code=start
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        if s.len() == 0 {
            return true;
        }
        if s.len() > t.len() {
            return false;
        }
        let s_bytes = s.as_bytes();
        let mut slow = 0;
        for fast in t.bytes() {
            if s_bytes[slow] == fast {
                slow += 1;
            }
            if slow == s.len() {
                return true;
            }
        }
        return false;
    }
}
// @lc code=end

File: javascript/392.is-subsequence.js
/*
 * @lc app=leetcode id=392 lang=javascript
 *
 * [392] Is Subsequence
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  if (!s.length) return true;
  if (s.length > t.length) return false;
  let i = 0;
  for (const c of t) {
    if (c === s[i]) i++;
  }
  return i === s.length;
};
// @lc code=end
